// accepts/index.js
cd C:\Users\HP\Documents\ACTIVIDADES
mkdir semana8
cd semana8
mkdir seguridad

Set-Content preguntas-semana8.txt @"
Pregunta 1:
Que es un sistema operativo?
Respuesta: B. El software que administra el hardware y permite ejecutar programas

Pregunta 2:
Cual de los siguientes es un sistema operativo de codigo abierto?
Respuesta: A. Ubuntu

Pregunta 3:
Que es la terminal en un sistema Linux?
Respuesta: B. Una interfaz que permite ejecutar comandos directamente en el sistema

Pregunta 4:
Cual de las siguientes practicas mejora la seguridad digital?
Respuesta: B. Usar contrasenas fuertes y proteger credenciales
"@

Set-Content seguridad\buenas-practicas.txt @"
Buenas practicas de seguridad digital

1. Proteger contrasenas usando combinaciones seguras de letras, numeros y simbolos.
2. No compartir credenciales con otras personas ni guardarlas en archivos publicos.
3. Mantener el software y los sistemas operativos siempre actualizados.
"@

Set-Content sistema.txt @"
Informacion del sistema — Comandos uname -a y whoami

Comando: uname -a
Este comando muestra informacion detallada sobre el sistema operativo: el nombre del
kernel, la version, la arquitectura del procesador y la fecha de compilacion. Es util
para saber exactamente que version de Linux se esta ejecutando en el servidor.

Comando: whoami
Este comando muestra el nombre del usuario que esta ejecutando la sesion actual en la
terminal. Es util para verificar con que permisos se esta trabajando, especialmente para
saber si se esta operando como usuario normal o como administrador (root).

Por que es importante conocer esta informacion al administrar un servidor:
Conocer el sistema operativo y el usuario activo es fundamental para tomar decisiones
correctas de seguridad y configuracion. Un administrador debe saber que version del
sistema tiene para aplicar actualizaciones de seguridad, y debe verificar los permisos
del usuario para evitar ejecutar comandos peligrosos con privilegios innecesarios.
"@

Set-Content seguridad-plataformas.txt @"
Buenas practicas de seguridad para plataformas digitales comunitarias

1. No exponer datos personales de los usuarios en respuestas publicas de la API ni en
   archivos accesibles sin autenticacion.

2. Proteger credenciales y claves de acceso usando variables de entorno en lugar de
   escribirlas directamente en el codigo fuente del proyecto.

3. Usar conexiones seguras (HTTPS) para que los datos transmitidos entre el cliente y el
   servidor viajen cifrados y no puedan ser interceptados.

4. Registrar accesos y actividad del sistema para detectar comportamientos sospechosos
   o intentos de acceso no autorizado a la plataforma.

5. Limitar el acceso a informacion sensible solo a usuarios o roles autorizados, aplicando
   controles de autenticacion y permisos dentro de la aplicacion.
"@

Set-Content reflexion-semana8.txt @"
Reflexion Semana 8

La seguridad digital es un pilar fundamental para cualquier plataforma tecnologica,
especialmente cuando esta es utilizada por comunidades o procesos participativos donde
se maneja informacion sensible de los ciudadanos. Una plataforma sin medidas de seguridad
adecuadas puede ser vulnerable a ataques que expongan datos personales, alteren reportes
o interrumpan el servicio, lo que afecta directamente la confianza de los usuarios.

En el contexto comunitario, la seguridad es aun mas importante porque las personas que
usan estas plataformas confian en que su informacion estara protegida. Si un sistema
comunitario sufre una brecha de seguridad, las consecuencias pueden ir mas alla de lo
tecnico: pueden afectar la reputacion de la iniciativa, desincentivar la participacion
ciudadana y poner en riesgo a personas vulnerables.

Practicas como el uso de contrasenas fuertes, conexiones cifradas y control de accesos
no son opcionales sino necesarias desde el inicio del desarrollo. Aprender sobre Linux y
seguridad digital desde etapas tempranas de formacion permite construir plataformas mas
solidas, responsables y confiables para las comunidades que las utilizan.
"@

cd C:\Users\HP\Documents\ACTIVIDADES
git add .
git commit -m "Actividad semana 8"
git push/*!
 * accepts
 * Copyright(c) 2014 Jonathan Ong
 * Copyright(c) 2015 Douglas Christopher Wilson
 * MIT Licensed
 */

'use strict'

/**
 * Module dependencies.
 * @private
 */

var Negotiator = require('negotiator')
var mime = rcd C:\Users\HP\Documents\ACTIVIDADES
mkdir semana8
cd semana8
mkdir seguridad

Set-Content preguntas-semana8.txt @"
Pregunta 1:
Que es un sistema operativo?
Respuesta: B. El software que administra el hardware y permite ejecutar programas

Pregunta 2:
Cual de los siguientes es un sistema operativo de codigo abierto?
Respuesta: A. Ubuntu

Pregunta 3:
Que es la terminal en un sistema Linux?
Respuesta: B. Una interfaz que permite ejecutar comandos directamente en el sistema

Pregunta 4:
Cual de las siguientes practicas mejora la seguridad digital?
Respuesta: B. Usar contrasenas fuertes y proteger credenciales
"@

Set-Content seguridad\buenas-practicas.txt @"
Buenas practicas de seguridad digital

1. Proteger contrasenas usando combinaciones seguras de letras, numeros y simbolos.
2. No compartir credenciales con otras personas ni guardarlas en archivos publicos.
3. Mantener el software y los sistemas operativos siempre actualizados.
"@

Set-Content sistema.txt @"
Informacion del sistema — Comandos uname -a y whoami

Comando: uname -a
Este comando muestra informacion detallada sobre el sistema operativo: el nombre del
kernel, la version, la arquitectura del procesador y la fecha de compilacion. Es util
para saber exactamente que version de Linux se esta ejecutando en el servidor.

Comando: whoami
Este comando muestra el nombre del usuario que esta ejecutando la sesion actual en la
terminal. Es util para verificar con que permisos se esta trabajando, especialmente para
saber si se esta operando como usuario normal o como administrador (root).

Por que es importante conocer esta informacion al administrar un servidor:
Conocer el sistema operativo y el usuario activo es fundamental para tomar decisiones
correctas de seguridad y configuracion. Un administrador debe saber que version del
sistema tiene para aplicar actualizaciones de seguridad, y debe verificar los permisos
del usuario para evitar ejecutar comandos peligrosos con privilegios innecesarios.
"@

Set-Content seguridad-plataformas.txt @"
Buenas practicas de seguridad para plataformas digitales comunitarias

1. No exponer datos personales de los usuarios en respuestas publicas de la API ni en
   archivos accesibles sin autenticacion.

2. Proteger credenciales y claves de acceso usando variables de entorno en lugar de
   escribirlas directamente en el codigo fuente del proyecto.

3. Usar conexiones seguras (HTTPS) para que los datos transmitidos entre el cliente y el
   servidor viajen cifrados y no puedan ser interceptados.

4. Registrar accesos y actividad del sistema para detectar comportamientos sospechosos
   o intentos de acceso no autorizado a la plataforma.

5. Limitar el acceso a informacion sensible solo a usuarios o roles autorizados, aplicando
   controles de autenticacion y permisos dentro de la aplicacion.
"@

Set-Content reflexion-semana8.txt @"
Reflexion Semana 8

La seguridad digital es un pilar fundamental para cualquier plataforma tecnologica,
especialmente cuando esta es utilizada por comunidades o procesos participativos donde
se maneja informacion sensible de los ciudadanos. Una plataforma sin medidas de seguridad
adecuadas puede ser vulnerable a ataques que expongan datos personales, alteren reportes
o interrumpan el servicio, lo que afecta directamente la confianza de los usuarios.

En el contexto comunitario, la seguridad es aun mas importante porque las personas que
usan estas plataformas confian en que su informacion estara protegida. Si un sistema
comunitario sufre una brecha de seguridad, las consecuencias pueden ir mas alla de lo
tecnico: pueden afectar la reputacion de la iniciativa, desincentivar la participacion
ciudadana y poner en riesgo a personas vulnerables.

Practicas como el uso de contrasenas fuertes, conexiones cifradas y control de accesos
no son opcionales sino necesarias desde el inicio del desarrollo. Aprender sobre Linux y
seguridad digital desde etapas tempranas de formacion permite construir plataformas mas
solidas, responsables y confiables para las comunidades que las utilizan.
"@

cd C:\Users\HP\Documents\ACTIVIDADES
git add .
git commit -m "Actividad semana 8"
git pushequire('mime-types')

/**
 * Module exports.
 * @public
 */

module.exports = Accepts

/**
 * Create a new Accepts object for the given req.
 *
 * @param {object} req
 * @public
 */

function Accepts (req) {
  if (!(this instanceof Accepts)) {
    return new Accepts(req)
  }

  this.headers = req.headers
  this.negotiator = new Negotiator(req)
}

/**
 * Check if the given `type(s)` is acceptable, returning
 * the best match when true, otherwise `undefined`, in which
 * case you should respond with 406 "Not Acceptable".
 *
 * The `type` value may be a single mime type string
 * such as "application/json", the extension name
 * such as "json" or an array `["json", "html", "text/plain"]`. When a list
 * or array is given the _best_ match, if any is returned.
 *
 * Examples:
 *
 *     // Accept: text/html
 *     this.types('html');
 *     // => "html"
 *
 *     // Accept: text/*, application/json
 *     this.types('html');
 *     // => "html"
 *     this.types('text/html');
 *     // => "text/html"
 *     this.types('json', 'text');
 *     // => "json"
 *     this.types('application/json');
 *     // => "application/json"
 *
 *     // Accept: text/*, application/json
 *     this.types('image/png');
 *     this.types('png');
 *     // => undefined
 *
 *     // Accept: text/*;q=.5, application/json
 *     this.types(['html', 'json']);
 *     this.types('html', 'json');
 *     // => "json"
 *
 * @param {String|Array} types...
 * @return {String|Array|Boolean}
 * @public
 */

Accepts.prototype.type =
Accepts.prototype.types = function (types_) {
  var types = types_

  // support flattened arguments
  if (types && !Array.isArray(types)) {
    types = new Array(arguments.length)
    for (var i = 0; i < types.length; i++) {
      types[i] = arguments[i]
    }
  }

  // no types, return all requested types
  if (!types || types.length === 0) {
    return this.negotiator.mediaTypes()
  }

  // no accept header, return first given type
  if (!this.headers.accept) {
    return types[0]
  }

  var mimes = types.map(extToMime)
  var accepts = this.negotiator.mediaTypes(mimes.filter(validMime))
  var first = accepts[0]

  return first
    ? types[mimes.indexOf(first)]
    : false
}

/**
 * Return accepted encodings or best fit based on `encodings`.
 *
 * Given `Accept-Encoding: gzip, deflate`
 * an array sorted by quality is returned:
 *
 *     ['gzip', 'deflate']
 *
 * @param {String|Array} encodings...
 * @return {String|Array}
 * @public
 */

Accepts.prototype.encoding =
Accepts.prototype.encodings = function (encodings_) {
  var encodings = encodings_

  // support flattened arguments
  if (encodings && !Array.isArray(encodings)) {
    encodings = new Array(arguments.length)
    for (var i = 0; i < encodings.length; i++) {
      encodings[i] = arguments[i]
    }
  }

  // no encodings, return all requested encodings
  if (!encodings || encodings.length === 0) {
    return this.negotiator.encodings()
  }

  return this.negotiator.encodings(encodings)[0] || false
}

/**
 * Return accepted charsets or best fit based on `charsets`.
 *
 * Given `Accept-Charset: utf-8, iso-8859-1;q=0.2, utf-7;q=0.5`
 * an array sorted by quality is returned:
 *
 *     ['utf-8', 'utf-7', 'iso-8859-1']
 *
 * @param {String|Array} charsets...
 * @return {String|Array}
 * @public
 */

Accepts.prototype.charset =
Accepts.prototype.charsets = function (charsets_) {
  var charsets = charsets_

  // support flattened arguments
  if (charsets && !Array.isArray(charsets)) {
    charsets = new Array(arguments.length)
    for (var i = 0; i < charsets.length; i++) {
      charsets[i] = arguments[i]
    }
  }

  // no charsets, return all requested charsets
  if (!charsets || charsets.length === 0) {
    return this.negotiator.charsets()
  }

  return this.negotiator.charsets(charsets)[0] || false
}

/**
 * Return accepted languages or best fit based on `langs`.
 *
 * Given `Accept-Language: en;q=0.8, es, pt`
 * an array sorted by quality is returned:
 *
 *     ['es', 'pt', 'en']
 *
 * @param {String|Array} langs...
 * @return {Array|String}
 * @public
 */

Accepts.prototype.lang =
Accepts.prototype.langs =
Accepts.prototype.language =
Accepts.prototype.languages = function (languages_) {
  var languages = languages_

  // support flattened arguments
  if (languages && !Array.isArray(languages)) {
    languages = new Array(arguments.length)
    for (var i = 0; i < languages.length; i++) {
      languages[i] = arguments[i]
    }
  }

  // no languages, return all requested languages
  if (!languages || languages.length === 0) {
    return this.negotiator.languages()
  }

  return this.negotiator.languages(languages)[0] || false
}

/**
 * Convert extnames to mime.
 *
 * @param {String} type
 * @return {String}
 * @private
 */

function extToMime (type) {
  return type.indexOf('/') === -1
    ? mime.lookup(type)
    : type
}

/**
 * Check if mime is valid.
 *
 * @param {String} type
 * @return {Boolean}
 * @private
 */

function validMime (type) {
  return typeof type === 'string'
}
